### PR TITLE
ICU-21107 Specify language standard versions C11 & C++17 also for MSVC

### PR DESCRIPTION
**Pull request by mihnita**
##### Checklist

- [x] Required: Issue filed: https://unicode-org.atlassian.net/browse/ICU-21107
- [x] Required: The PR title must be prefixed with a JIRA Issue number. 
- [x] Required: The PR description must include the link to the Jira Issue, for example by completing the URL in the first checklist item
- [x] Required: Each commit message must be prefixed with a JIRA Issue number. 
- [x] Issue accepted (done by Technical Committee after discussion)
- [x] Tests included, if applicable
- [x] API docs and/or User Guide docs changed or added, if applicable

### CODE REVIEW

**Review by eggrobin — DISMISSED** [review]
Tested that I was able to change https://github.com/unicode-org/icu/blob/ab72ab1d4a3c3f9beeb7d92b0c7817ca93dfdb04/icu4c/source/test/intltest/usettest.cpp#L1427-L1429

to
```C++
for (auto [c, t] : closure) {
```

and that it compiled.

**Comment by eggrobin** [comment]
Ah right, there is that TestCalendarOperations issue here too…

File: icu4c/source/i18n/calendar.cpp [hunk 1]
```diff
@@ -3695,13 +3695,15 @@ int32_t Calendar::handleGetExtendedYearFromWeekFields(int32_t yearWoy, int32_t w
 
 int32_t Calendar::handleGetMonthLength(int32_t extendedYear, int32_t month) const
 {
-    return handleComputeMonthStart(extendedYear, month+1, true) -
-        handleComputeMonthStart(extendedYear, month, true);
+    const int32_t crtYear = handleComputeMonthStart(extendedYear, month+1, true);
+    const int32_t nextYear = handleComputeMonthStart(extendedYear, month, true);
+    return crtYear - nextYear;
```
Comment (eggrobin) [inline icu4c/source/i18n/calendar.cpp:3700]: https://github.com/unicode-org/icu/pull/2767/commits/d5bf628527850f446be20063c11c63b1b9b83695 only fixed handleGetYearLength, it seems like a good idea fix this one too. Though you will want these to be called `Month`, and I think you have current and next swapped.

Comment (mihnita) [reply icu4c/source/i18n/calendar.cpp:3700]: I know :-) Done already.
I was also working to solve the conflicts, but I was running late (missing the bus)
Thanks,
M

**Review by eggrobin — APPROVED** [review]

**Pull request merged**